src/test/controller/UserController.test.ts

```diff
@@ -363,12 +422,13 @@ describe('Unit tests for UserController', function () {
       });
   });
 
-  test('GET /users/:id/uploadedPromotions - should return empty list if user has no uploaded promotions', async (done) => {
+  xtest('GET /users/:id/uploadedPromotions - should return empty list if user has no uploaded promotions', async (done) => {
```

Comment:
You're getting deadlocks for uploadedPromotions endpoints? I was only seeing deadlocks for savedPromotions

Reply:
This is because the deadlock happened once on this function, but it would be fixed so changed to test